radialgraph/src/main/java/com/duartbreedt/radialgraph/drawable/GraphDrawable.kt

```diff
@@ -29,11 +34,67 @@ abstract class GraphDrawable(
 
         return Paint().apply {
             strokeWidth = graphConfig.strokeWidth
-            color = state.color
             pathEffect = DashPathEffect(floatArrayOf(state.length!!, state.length!!), phase)
             style = Paint.Style.STROKE
             flags = Paint.ANTI_ALIAS_FLAG
             strokeCap = graphConfig.capStyle.paintCapStyle
+
+            if (!graphConfig.isGradientEnabled()) {
+                color = state.color
+            }
+
+            if (graphConfig.gradientType == GradientType.SWEEP) {
+                applySweepGradient(state, this, graphConfig.gradientFill)
+            }
+        }
+    }
+
+    private fun applySweepGradient(state: SectionState, paint: Paint, gradientFill: GradientFill) {
+        if (state.gradientColors.isNullOrEmpty()) {
+            paint.color = state.color
+            return
+        }
+
+        val colorList = mutableListOf(
+            state.color
+        )
+        colorList.addAll(state.gradientColors)
+
+        var positionList: MutableList<Float>? = null
+        if (gradientFill == GradientFill.SECTION) {
+            positionList = MutableList(colorList.size) { it.toFloat() }
+            //Get spacing between each gradient section
+            val gradientGap = (1f / (positionList.size -1)) * state.sweepSize
+
+            positionList = positionList.map {
+                (gradientGap * it) + state.startPosition
+            }.toMutableList()
+
+            //Fix for gradient overflow when using a cap style other than 'BUTT'
+            if (state.startPosition == 0f && graphConfig.capStyle != Cap.BUTT) {
+                colorList.add(state.color)
+                positionList.addAll(
+                    listOf(
+                        positionList.removeLast().coerceAtMost(endGradientCapFill),
+                        endGradientCapFill
+                    )
+                )
```

Comment:
What this does is not clear to me, please can you elaborate on this logic

Reply:
This is a fix for when the cap style is `ROUNDED` or `SQUARE`, it'll draw the last color on the initial cap because that cap overflows to < 0.

This will set the start color right at the end and set a position for it so that it will look right. (It will also adjust the ending gradient position)

Without Fix:
<img src="https://github.com/DuartBreedt/RadialGraph/assets/47508305/e2ace1cc-b8bf-4b46-b963-009962aff061" width="200px" />


With Fix:
<img src="https://github.com/DuartBreedt/RadialGraph/assets/47508305/5fe134da-d9f5-4c58-a327-085d5e3ace85" width="200px" />